src/Symfony/Component/DependencyInjection/Attribute/Autowire.php

```diff
@@ -33,15 +33,15 @@ class Autowire
      * @param string|null $expression Expression (ie 'service("some.service").someMethod()')
      */
     public function __construct(
-        string $value = null,
+        string|array $value = null,
         string $service = null,
         string $expression = null,
     ) {
         if (!($service xor $expression xor null !== $value)) {
             throw new LogicException('#[Autowire] attribute must declare exactly one of $service, $expression, or $value.');
         }
 
-        if (null !== $value && str_starts_with($value, '@')) {
+        if (\is_string($value) && str_starts_with($value, '@')) {
```

Comment:
For an array, wouldn't this need to process the strings inside that array to convert their prefixes too ?

Reply:
I don't think we want to allow nesting services into arrays.

Reply:
We do allow that in Yaml or XML (which is how you passed an array of objects indexed by some keys for instance)